cli/command_snapshot_create.go

```diff
@@ -168,6 +177,31 @@ func snapshotSingleSource(ctx context.Context, rep repo.RepositoryWriter, u *sna
 		return errors.Wrap(err, "unable to get local filesystem entry")
 	}
 
+	return snapshotSource(ctx, rep, u, sourceInfo, localEntry, false)
+}
+
+func snapshotStdinSource(ctx context.Context, rep repo.RepositoryWriter, u *snapshotfs.Uploader) error {
+	rootDirName := *snapshotCreateStdinDirName
+	stdinFileName := *snapshotCreateStdinFileName
+
+	log(ctx).Infof("Snapshotting stdin source as %v ...", rootDirName)
+
+	// stdin source will be snapshotted using a virtual static root directory with a single streaming file entry
+	// Create a new static directory with the given name and add a streaming file entry with os.Stdin reader
+	staticDir := virtualfs.NewStaticDirectory(rootDirName, fs.Entries{
```

Comment:
I think we can simplify the flags a bit here:

What do you think of removing --rootdir to simplify the UX like this:

```
$ kopia snapshot create /some/dir --stdin-file=filename ?
```

Also, how about this:

```
$ kopia snapshot create /some/file --stdin
```

(this could create a single-file snapshot, not a directory-with-a-single-file snapshot, this is optional, could be in separate PR).

Reply:
Note that to support this, I have changed the args to be a list of `Strings()`. The snapshot create command will now fail at a later stage if the provided source file or directory does not exist